src/realm/object-store/sync/sync_session.cpp

```diff
@@ -97,37 +96,36 @@ void SyncSession::become_active(std::unique_lock<std::mutex>& lock)
     std::swap(m_completion_callbacks, callbacks_to_register);
 
     for (auto& [id, callback_tuple] : callbacks_to_register) {
-        add_completion_callback(lock, std::move(callback_tuple.second), callback_tuple.first);
+        add_completion_callback(std::move(callback_tuple.second), callback_tuple.first);
     }
 }
 
-void SyncSession::become_dying(std::unique_lock<std::mutex>& lock)
+void SyncSession::become_dying(util::CheckedUniqueLock lock)
```

Comment:
The ones which actually do need the lock now take it by value instead of by reference. This makes it so that the mutex is always unlocked when we exit the function, rather than leaving it locked if we happen to throw an exception prior to reaching the point where we unlock it.